File: c#-solution/0476. Number Complement.cs
// 0476. Number Complement
// https://leetcode.com/problems/number-complement/
// Tag: C#, Bit Manipulation, Easy

// string
public class Solution {
    public int FindComplement(int num) {
        var n = Convert.ToString(num, 2);
        var s = new StringBuilder();
        foreach(char i in n){
            var c = i =='1' ? '0' : '1' ;
            s.Append(c);
        }
        return Convert.ToInt32(s.ToString(), 2);
    }
}

File: problem/0476. Number Complement.js
// 476. Number Complement
// https://leetcode.com/problems/number-complement/
// Tag: Bit Manipulation, Easy

/**
 * @param {number} num
 * @return {number}
 */


var findComplement = function(num) {
    let bi = num.toString(2);
  
    let complement = '';
    for(let i=0; i<bi.length; i++){
        complement += bi[i] === "1"? "0":"1";
    }
    return parseInt(complement, 2);
};